### PR TITLE
[Coop/icalls]  Convert a few, nohandles() a few, cleanup a few.

### PR DESCRIPTION
**Pull request by jaykrell**
[Coop] Convert ves_icall_System_Object_MemberwiseClone.
[Coop] Convert ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_GetObjectValue.
[Coop] Convert ves_icall_MonoField_get_core_clr_security_level.
[Coop] Convert ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_RunClassConstructor.
[Coop] Convert ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_RunModuleConstructor.
[coop] Remove redundant declarations.
[Coop] Nohandles ves_icall_System_Environment_Exit.
[Coop] Nohandles ves_icall_System_Environment_GetIs64BitOperatingSystem.
[Coop] Nohandles mono_pagesize.
[Coop] Nohandles mono_environment_exitcode_get.
[Coop] Nohandles ves_icall_System_Environment_get_HasShutdownStarted.
[Coop] Nohandles ves_icall_System_Environment_get_Platform.
[Coop] Nohandles ves_icall_System_Environment_get_ProcessorCount.
[Coop] Nohandles ves_icall_System_Environment_get_TickCount.
[Coop] Nohandles mono_environment_exitcode_set.
[Coop] Nohandles ves_icall_System_Reflection_AssemblyName_GetNativeName.
[Coop] Nohandles ves_icall_System_Reflection_AssemblyName_ParseAssemblyName.
[Coop] Nohandles mono_digest_get_public_token.
[Coop] Nohandles ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_GetOffsetToStringData.
[icalls] Cleanup ves_icall_System_Environment_get_HasShutdownStarted.
[icalls] Cleanup ves_icall_System_Environment_BroadcastSettingChange.
[icalls] Cleanup ves_icall_IsTransparentProxy.
[icalls] Cleanup ves_icall_Remoting_RemotingServices_GetVirtualMethod.
[icalls] Cleanup ves_icall_System_Runtime_Activation_ActivationServices_EnableProxyActivation.
[icalls] Cleanup ves_icall_System_Runtime_Activation_ActivationServices_AllocateUninitializedClassInstance.
[icalls] Cleanup ves_icall_RuntimeMethodHandle_GetFunctionPointer.
[icalls] Cleanup ves_icall_System_Configuration_DefaultConfig_get_machine_config_path.
[icalls] Cleanup ves_icall_System_Configuration_InternalConfigurationHost_get_bundled_app_config.
[icalls] Cleanup ves_icall_System_Web_Util_ICalls_get_machine_install_dir.
[icalls] Cleanup ves_icall_get_resources_ptr
[icalls] Cleanup ves_icall_System_Diagnostics_Debugger_IsLogging.
[icalls] Cleanup ves_icall_System_Activator_CreateInstanceInternal.
[icalls] Cleanup ves_icall_MonoMethod_get_base_method.
[icalls] Cleanup ves_icall_MonoMethod_get_name.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

**Comment by jaykrell** [comment]
@monojenkins build OS  X  x64 C++

**Comment by jaykrell** [comment]
@monojenkins build Linux ARMv7

**Comment by jaykrell** [comment]
@monojenkins build OS X x64 C++

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -974,37 +968,30 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_RunClassConstructor (Mo
...
-			return;
-		}
-		if (!mono_runtime_class_init_full (vtable, error))
-			mono_error_set_pending_exception (error);
-	}
+	if (!image->has_module_cctor)
+		return;
+
+	MonoClass *module_klass = mono_class_get_checked (image, MONO_TOKEN_TYPE_DEF | 1, error);
+	if (!mono_error_ok (error))
```
Comment (lambdageek) [inline mono/metadata/icall.c]: could use `is_ok`

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -974,37 +968,30 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_RunClassConstructor (Mo
...
+		return;
+
+	/*It's fine to raise the exception here*/
```
Comment (lambdageek) [inline mono/metadata/icall.c]: delete this comment - it's not relevant anymore

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Windows x64

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
This will probably break wasm/cxx again due to slight mismerge.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build  Linux AArch64 Coop Suspend

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 Coop Suspend

**Pull request merged**